docs/layering.md

```diff
@@ -0,0 +1,367 @@
+tutorial login: core (automatic login)
+
+[core@tutorial ~]$
+```
+
+We can verify that our configuration has been applied: we were automatically logged in to the terminal, and the hostname on the prompt is `tutorial`.
+
+As we can see our `/etc/cachefilesd.conf` file:
+
+```
+[core@tutorial ~]$ cat /etc/cachefilesd.conf
+dir /var/cache/fscache
+tag mycache
+brun  8%
+bcull 6%
+bstop 2%
+frun  9%
+fcull 5%
+fstop 4%
+secctx system_u:system_r:cachefiles_kernel_t:s0
+```
+
+Let's also make sure there is no cachefiled package installed.
+```
+[core@tutorial ~]$ rpm -qi cachefilesd
+package cachefilesd is not installed
+```
+
+### Create a derived image with additional content
+
+We are assuming there is a need for debugging on your environment and you need strace and a custom binary. Finally we are going to add keylime and rebase your running FCOS deployment to include them.
+
+This would be helpful when IT needs to provide the images or there is a need to add the same system image to multiple hosts. 
+Instead of making the changes on each host which will have different versions of software or configs you can create an image which then you can rebase your systems on. We are using the [fcos-derivation-example](https://github.com/coreos/fcos-derivation-example) to start this process.
+
+Let's clone our example code and enter the directory.
+
+```bash
+[core@tutorial ~]$ git clone git@github.com:coreos/fcos-derivation-example.git && cd fcos-derivation-example
+```
+
+Examine the Dockerfile and notice the custom binary and where we are installing strace.
+
+```bash
+$ vi Dockerfile
+```
+
+Let's add `cachefilesd` to the install command.
+
+That would make the last line look like:
+```dockerfile
+RUN rpm-ostree install strace cachefilesd && rpm-ostree cleanup -m
+```
+
+Time to build the image. In this example we are using podman.
+```
+$ podman build -t localhost/my-custom-fcos .
+```
+
+### Inspect the content of the derived image
+At this point we have an OCI image. You can test to see if it actually has the layered packages. 
+
+Let's get into the container.
+```
+$ podman run -it localhost/my-custom-fcos /bin/bash
+```
+
+Run stace inside the container. You should see the following:
+```bash
+$ strace -- version 5.15
+
+Copyright (c) 1991-2021 The strace developers <https://strace.io>.
+This is free software; see the source for copying conditions.  There is NO
+warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+Optional features enabled: stack-trace=libdw stack-demangle m32-mpers mx32-mpers secontext
+```
+
+Which tells us we have successfully installed `strace` on Fedora CoreOS.
+
+Next let's see if cachefilesd is there.
+```bash
+$ cachefilesd --version     
+```
+
+The output should look like the following:
+```
+cachefilesd version 0.10.10
+```
+
+Now that we confirmed our packages are there, we can get out of the container:
+```bash
+$ exit
+```
+
+### Push the OCI image to a container registry
```

Comment:
You can also directly build on Quay.io and pull the image from there. It's much less expensive than pushing for some connections. I don't know if we want to add that here.

Reply:
I could add a link to https://docs.quay.io/guides/building.html and so people know is an option. WDYT?

Reply:
added info about this feature.